secure/storage/src/tests/suite.rs

```diff
@@ -1,110 +1,180 @@
+        vault::reset_vault_storage(storage.as_mut());
+        test(storage.as_mut());
+    }
+}
 
-    let mut rng = StdRng::from_seed([13u8; 32]);
-    let key_value_0 = Ed25519PrivateKey::generate_for_testing(&mut rng);
-    let key_value_1 = Ed25519PrivateKey::generate_for_testing(&mut rng);
+/// This test tries to get and set non-existent keys in storage and asserts that the correct
+/// errors are returned on these operations.
+fn test_get_set_non_existent(storage: &mut dyn Storage) {
+    let crypto_key = "Private Key";
+    let crypto_value = Value::Ed25519PrivateKey(create_ed25519_key_for_testing());
+    let u64_key = "U64 Key";
+    let u64_value = Value::U64(10);
 
     assert_eq!(
-        storage.get(KEY_KEY).unwrap_err(),
-        Error::KeyNotSet(KEY_KEY.to_string())
+        storage.get(crypto_key).unwrap_err(),
+        Error::KeyNotSet(crypto_key.to_string())
     );
     assert_eq!(
-        storage.get(U64_KEY).unwrap_err(),
-        Error::KeyNotSet(U64_KEY.to_string())
+        storage.get(u64_key).unwrap_err(),
+        Error::KeyNotSet(u64_key.to_string())
     );
-
     assert_eq!(
-        storage
-            .set(KEY_KEY, Value::Ed25519PrivateKey(key_value_0.clone()))
-            .unwrap_err(),
-        Error::KeyNotSet(KEY_KEY.to_string())
+        storage.set(crypto_key, crypto_value).unwrap_err(),
+        Error::KeyNotSet(crypto_key.to_string())
     );
     assert_eq!(
-        storage.set(U64_KEY, Value::U64(u64_value_0)).unwrap_err(),
-        Error::KeyNotSet(U64_KEY.to_string())
+        storage.set(u64_key, u64_value).unwrap_err(),
+        Error::KeyNotSet(u64_key.to_string())
     );
+}
+
+/// This test stores various key/value pairs in storage, updates them, retrieves the values and
+/// unwraps them to ensure the correct value types are returned.
+fn test_create_get_set_unwrap(storage: &mut dyn Storage) {
+    let crypto_key = "Private Key";
+    let crypto_private_1 = create_ed25519_key_for_testing();
+    let crypto_private_2 = create_ed25519_key_for_testing();
+    let crypto_value_1 = Value::Ed25519PrivateKey(crypto_private_1.clone());
+    let crypto_value_2 = Value::Ed25519PrivateKey(crypto_private_2.clone());
+
+    let u64_key = "U64 Key";
+    let u64_1 = 10;
+    let u64_2 = 647;
+    let u64_value_1 = Value::U64(u64_1);
+    let u64_value_2 = Value::U64(u64_2);
+
+    let policy = Policy::public();
 
     storage
-        .create_if_not_exists(U64_KEY, Value::U64(u64_value_1), &public)
-        .unwrap();
-    storage
-        .create(
-            KEY_KEY,
-            Value::Ed25519PrivateKey(key_value_1.clone()),
-            &public,
-        )
+        .create_if_not_exists(u64_key, u64_value_1, &policy)
         .unwrap();
+    storage.create(crypto_key, crypto_value_1, &policy).unwrap();
 
-    assert_eq!(storage.get(U64_KEY).unwrap().u64().unwrap(), u64_value_1);
+    assert_eq!(storage.get(u64_key).unwrap().u64().unwrap(), u64_1);
     assert_eq!(
```

Comment:
if we do this, do we need the u64_value_* any more? seems unnecessary.

Reply:
Ooh, nice spot! Thanks!